tests/cephadm/test_cephadm_pull.py

```diff
@@ -0,0 +1,61 @@
+import requests
+import yaml
+
+from ceph.ceph_admin import CephAdmin
+from utility.log import Log
+
+log = Log(__name__)
+
+
+class CephadmpullFailure(Exception):
+    pass
+
+
+def _get_latest_ceph_version(url):
+    """
+    Get latest ceph version from URL
+    Args:
+        url: Ceph container yaml link
+    Returns:
+        Latest ceph version
+    """
+
+    response = requests.get(url)
+    yaml_response = yaml.safe_load(response.text)
+    log.info(f"Ceph versions available in provided yaml link {yaml_response}'")
+    ceph_versions = list(yaml_response.keys())
+    ceph_versions.sort()
+    return yaml_response[ceph_versions[-1]]["ceph-version"]
+
+
+def run(ceph_cluster, **kw):
+    """
+    Verify ceph version using cephadm pull option
+    Args:
+        ceph_cluster: Ceph cluster object
+        **kw :     Key/value pairs of configuration information to be used in the test.
+
+    Returns:
+        int - 0 when the execution is successful.
+    """
+
+    config = kw.get("config")
+    # Executing cephadm pull command
+    instance = CephAdmin(cluster=ceph_cluster, **config)
+    # Geting ceph version detail using cephadm pull command
+    cmd = "cephadm pull"
```

Comment:
```
[cephuser@ceph-prag00-20qemv-node1-installer ~]$ sudo cephadm pull
Pulling container image quay.io/ceph/ceph:v17...
{
    "ceph_version": "ceph version 17.2.3 (dff484dfc9e19a9819f375586300b3b79d80034d) quincy (stable)",
    "image_id": "0912465dcea5159f56c7d4ccdf1d15a28abfe1be8a56c94eb86cab129c069726",
    "repo_digests": [
        "quay.io/ceph/ceph@sha256:43f6e905f3e34abe4adbc9042b9d6f6b625dee8fa8d93c2bae53fa9b61c3df1a",
        "quay.io/ceph/ceph@sha256:e4abc23698c84996b04a5bf7706ed7e8cc88b9b716dc98cd83681b178aa7631f"
    ]
}
[cephuser@ceph-prag00-20qemv-node1-installer ~]$ sudo podman image list
REPOSITORY                                                  TAG                                                        IMAGE ID      CREATED        SIZE
registry-proxy.engineering.redhat.com/rh-osbs/rhceph        ceph-6.0-rhel-9-containers-candidate-67315-20220822210801  e52607848380  11 hours ago   1.01 GB
quay.io/ceph/ceph                                           v17                                                        0912465dcea5  2 weeks ago    1.38 GB
registry.redhat.io/openshift4/ose-prometheus-alertmanager   v4.9.0-202110182323.p0.git.579e3c6.assembly.stream         6956d220b8fc  10 months ago  306 MB
registry.redhat.io/openshift4/ose-prometheus                v4.9.0-202110182323.p0.git.3197fa7.assembly.stream         9124c6a4d9b6  10 months ago  403 MB
registry.redhat.io/rhceph/rhceph-5-dashboard-rhel8          5-27                                                       09cf77100f6a  12 months ago  506 MB
registry.redhat.io/openshift4/ose-prometheus-node-exporter  v4.6.0-202107070256.p0.git.c63b8f3                         ab5d6044e082  13 months ago  347 MB

```
This is the output that I saw in RHCS 6.0... the purpose of `cephadm pull` is to download all the latest images pertaining to Ceph. However, what is observed is that it is pulling images from upstream.

I believe verifying the version isn't sufficient and we need to check if the right images are being pulled.

Reply:
IMO, we should use the ceph version value that was displayed by `cephadm pull` and then refer the recipe file to validate if the right image tag for `rh-osbs/rhceph` was pulled

Reply:
@psathyan 
Thanks for the review comments.

As discussed
- Moved test case to test suite "tier-1_service_apply_spec.yaml" which is the part of pipeline.
- Removed hard cored "url" and added new method "fetch_image_tag" in util.
- Now comparing the image from podman image list(Executed after cephadmpull) with image from recipe file.
- Not comparing ceph version.